app/src/main/java/org/fossasia/susi/ai/device/deviceconnect/DeviceConnectFragment.kt

```diff
@@ -116,37 +124,198 @@ class DeviceConnectFragment : Fragment(), IDeviceConnectView {
+            }
+
+            override fun onTextChanged(charSequence: CharSequence, start: Int, before: Int, count: Int) {
+                // If password input is not empty then make the finish button enable else disable it and change the colour accordingly
+                if (charSequence.trim().length > 0) {
+                    password_finish.isEnabled = true
+                    password_finish.isClickable = true
+                    password_finish.background = ContextCompat.getDrawable(requireContext(), R.drawable.border_blue)
+                    password_finish.setTextColor(resources.getColor(R.color.md_white_1000))
+                } else {
+                    password_finish.isEnabled = false
+                    password_finish.isClickable = false
+                    password_finish.background = ContextCompat.getDrawable(requireContext(), R.drawable.border_normal)
+                    password_finish.setTextColor(resources.getColor(R.color.blue_grey_300))
+                }
+            }
+        }
+        account_auth_password_input.addTextChangedListener(watch)
+
+        // Action of previous button
+        password_previous.setOnClickListener {
+            rooms()
+        }
+
+        // Action of finish button
+        password_finish.setOnClickListener {
+            if (password_finish.isEnabled && password_finish.isClickable) {
+                // Send the password request
+                finishSetup()
+            }
+        }
+    }
+
+    override fun finishSetup() {
+        val dialogBuilder = AlertDialog.Builder(requireContext())
+        dialogBuilder.setMessage(getString(R.string.finish_setup_details))
+                .setCancelable(false)
+                .setPositiveButton(getString(R.string.finish_setup_button), DialogInterface.OnClickListener { dialog, id ->
+                    deviceConnectPresenter.makeAuthRequest(account_auth_password_input.text.toString())
+                    ANONYMOUS_MODE = false
+                    dialog.cancel()
+                })
+                .setNegativeButton(getString(R.string.go_back), DialogInterface.OnClickListener { dialog, id ->
+                    dialog.cancel()
+                }).setTitle(getString(R.string.finish_setup_title))
+                .create()
+                .show()
+    }
+
+    override fun successSetup() {
+        password_layout.visibility = View.GONE
+        success_setup_screen.visibility = View.VISIBLE
+        if (ANONYMOUS_MODE) {
+            success_message.text = getString(R.string.success_setup_anonymous)
+        } else {
+            success_message.text = getString(R.string.succesfully_setup)
+        }
+        unregister()
+        context?.registerReceiver(receiverWifi, IntentFilter())
+
+        success_setup.setOnClickListener {
+            val intent = Intent(context, SkillsActivity::class.java)
+            intent.putExtra(SETTINGS_FRAGMENT, SETTINGS_FRAGMENT)
```

Comment:
Why key and value are same?

Reply:
This is because the skills activity uses the name ```settingsFragment``` to inflate the SettingsFragment. The key and value there is the same. This was present for the past, so just re-used it here.

Reply:
You can use `redirectedFrom` as key and `settingsFragment` as value